### ISSUE TITLE
spark: `LogicalRelation` as an input dataset

### ISSUE THREAD

**Issue by wslulciuc** [OpenLineage/OpenLineage#838]
This issue is similar to issue #610. I wanted to start the discussion on how / if we wanted to handle [`org.apache.spark.sql.catalyst.plans.logical.LocalRelation`](https://jaceklaskowski.gitbooks.io/mastering-spark-sql/content/spark-sql-LogicalPlan-LocalRelation.html) when using spark SQL.

**Comment by collado-mike** [comment]
Is there a specific `BaseRelation` you wanted to discuss? We do have `LogicalRelationDatasetBuilder` already.

**Comment by pawel-big-lebowski** [comment]
tl;dr; Is there any real-life example for this feature? 

The example for `InMemoryRelation` is: we read data from table A, we cache it in the mean time, then we read the cached data and save as a new dataset. At the end we create OpenLineage event and would love to provide it with information that table A was an input dataset. That totally makes sense.

I saw `LocalRelation` in Spark's plans when data was provided by hand like this:
```
val someDF = Seq(
  (8, "bat"),
  (64, "mouse"),
  (-27, "horse")
).toDF("number", "word")
```
I did not think this would add any value to OL event. 
The point here is to understand why would anybody need this information. 
If there is no such reason, we can filter this out from `UnknownEntryFacet` as the entry is known but not needed.

**Comment by tnazarew** [comment]
I think this one can be closed. @pawel-big-lebowski ?
